project_1/include/event.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "server.hpp"
+
+#include <cstdint>
+#include <iostream>
+
+enum event_type: std::uint8_t {
+    SEND_DATA,
+    ACKNOWLEDGEMENT,
+};
+
+// Struct represents event -- sending data or acknowledgement
+struct Event {
```

Comment:
A class definition should usually start with a public: section, followed by protected:, then private:.
https://google.github.io/styleguide/cppguide.html#Declaration_Order